Analyzer/src/AnalyzeLepTrigger.cc

```diff
@@ -109,356 +97,158 @@ void AnalyzeLepTrigger::Loop(NTupleReader& tr, double weight, int maxevents, boo
-            
+            const auto& puWeight = tr.getVar<double>("puWeightCorr");
+
             // Define lepton weight
             if(NGoodLeptons == 1)
             {
                 const auto& eleLepWeight = tr.getVar<double>("totGoodElectronSF");
                 const auto& muLepWeight  = tr.getVar<double>("totGoodMuonSF");
                 leptonScaleFactor = (GoodLeptons[0].first == "e") ? eleLepWeight : muLepWeight;
             }
+
+            const auto& bTagScaleFactor   = tr.getVar<double>("bTagSF_EventWeightSimple_Central");
+            const auto& htDerivedScaleFactor = tr.getVar<double>("htDerivedweight");
+            const auto& topPtScaleFactor = tr.getVar<double>("topPtScaleFactor");
+            const auto& prefiringScaleFactor = tr.getVar<double>("prefiringScaleFactor");
+
+            eventweight = lumi*Weight*puWeight;
             
-            //PileupWeight = tr.getVar<double>("_PUweightFactor");
-            bTagScaleFactor   = tr.getVar<double>("bTagSF_EventWeightSimple_Central");
-            htDerivedScaleFactor = tr.getVar<double>("htDerivedweight");
-            topPtScaleFactor = tr.getVar<double>("topPtScaleFactor");
-            prefiringScaleFactor = tr.getVar<double>("prefiringScaleFactor");
-            puScaleFactor = tr.getVar<double>("puWeightCorr");
-            
-            weight *= eventweight*leptonScaleFactor*bTagScaleFactor*htDerivedScaleFactor*prefiringScaleFactor*puScaleFactor;
         }
-       
-        //Define trigger cuts
-        std::vector<std::string> myMuonTriggers             { "HLT_IsoMu24_v", "HLT_IsoTkMu24_v", "HLT_Mu50_v", "HLT_TkMu50_v" };
-        std::vector<std::string> myElectronTriggers         { "HLT_Ele27_WPTight_Gsf_v", "HLT_Photon175_v", "HLT_Ele115_CaloIdVT_GsfTrkIdT_v" };
 
+        //Define trigger cuts
+        std::vector<std::string> myMuonTriggers, myElectronTriggers;
+        if (runYear == "2016") {
+            std::vector<std::string> myMuonTriggers             { "HLT_IsoMu24_v", "HLT_IsoTkMu24_v", "HLT_Mu50_v", "HLT_TkMu50_v" };
+            std::vector<std::string> myElectronTriggers         { "HLT_Ele27_WPTight_Gsf_v", "HLT_Photon175_v", "HLT_Ele115_CaloIdVT_GsfTrkIdT_v" };
 
-        if( runYear == "2017" ) {
+        } else if( runYear == "2017" ) {
             std::vector<std::string> myMuonTriggers         { "HLT_IsoMu24", "HLT_IsoMu27", "HLT_Mu50", "HLT_TkMu50" };
-            std::vector<std::string> myElectronTriggers     { "HLT_Ele35_WPTight_Gsf", "HLT_Photon200", "HLT_Ele115CaloIdVT_GsfTrkIdT" };
+            std::vector<std::string> myElectronTriggers     { "HLT_Ele35_WPTight_Gsf", "HLT_Photon200", "HLT_Ele115_CaloIdVT_GsfTrkIdT" };
+        } else if (runYear == "2018") {
+            std::vector<std::string> myMuonTriggers         { "HLT_IsoMu24", "HLT_IsoMu27", "HLT_Mu50" };
+            std::vector<std::string> myElectronTriggers     { "HLT_Ele35_WPTight_Gsf", "HLT_Photon200", "HLT_Ele115_CaloIdVT_GsfTrkIdT" };
```

Comment:
Do you know for sure these triggers will be used for 2018 or is figuring that out part of your study?

Reply:
This still needs to be verified. To be confirmed.